demos/applications/apa102.cpp

```diff
@@ -57,66 +57,71 @@ void application(resource_list& p_map)
   using namespace std::chrono_literals;
   using namespace hal::literals;
 
+  // variables for LEDs
+  // led_count can be changed to customize demo
+  constexpr std::size_t led_count = 4;
+  hal::display::apa102_frame<led_count> apa_frame;
+  unsigned brightness = 1;
+
   // get resources
   auto& clock = *p_map.clock.value();
   auto& console = *p_map.console.value();
   auto& chip_select = *p_map.spi_chip_select.value();
   auto& spi = *p_map.spi.value();
 
-  // variables for LEDs
-  constexpr std::size_t led_count = 4;
-  hal::display::apa102_frame<led_count> apa_frame;
-  unsigned brightness = 1;
-
-  // predefined colors
-  hal::display::apa102_pixel red{ .blue = 0x00, .green = 0x00, .red = 0xFF };
-  hal::display::apa102_pixel green{ .blue = 0x00, .green = 0xFF, .red = 0x00 };
-  hal::display::apa102_pixel blue{ .blue = 0xFF, .green = 0x00, .red = 0x00 };
-  hal::display::apa102_pixel white{ .blue = 0xFF, .green = 0xFF, .red = 0xFF };
+  // predefined color arrays
   std::array<hal::display::apa102_pixel, led_count> all_off{};
-
-  std::array<hal::display::apa102_pixel, led_count> rgb_array = {
-    { white, blue, green, red }
+  std::array<hal::display::apa102_pixel, 4> predefined_colors = {
```

Comment:
```suggestion
  constexpr std::array<hal::display::apa102_pixel, 4> predefined_colors = {
```
The code doesn't update this value and its completely defined at compile time, so we can mark it as `constexpr`.